src/client/dfs/duns.c

```diff
@@ -373,39 +424,93 @@ duns_resolve_path(const char *path, struct duns_attr_t *attr)
 		return err;
 	}
 
+	realp = realpath(path, NULL);
+	if (realp == NULL)
+		return errno;
+
+	path_len = strnlen(realp, PATH_MAX);
+	if (path_len > PATH_MAX - 1)
+		D_GOTO(out, rc = ENAMETOOLONG);
+
+	D_ALLOC(rel_path, path_len + 1);
+	if (rel_path == NULL)
+		D_GOTO(out, rc = ENOMEM);
+
+	D_STRDUP(dir_path, realp);
```

Comment:
Since the length is calculated above, could we just do this?
```suggestion
	D_STRNDUP(dir_path, realp, path_len);
```

Reply:
I thought we were generally supposed to stay away from non-length versions of the string functions as much as possible, but I guess that isn't "official". If no one else complains about this, I'm okay with it :)

Reply:
@daltonbohning @mchaarawi I find it confusing when it is ok and when it is not as welll. Is it ok to use strdup if we know the length?